blockprod/src/lib.rs

```diff
@@ -89,3 +91,71 @@ pub async fn make_blockproduction(
 
     Ok(Box::new(result))
 }
+
+#[allow(clippy::unused_async)]
+pub async fn setup_blockprod_test() -> (Manager, Arc<ChainConfig>, ChainstateHandle, MempoolHandle)
```

Comment:
Does it really have to be exposed to non-test code? Maybe if many tests across the package require a complex setup all these tests can be moved to a separate module.

Also it should fix `pub` access level.

Reply:
Yeah I wasn't too sure about this one, but I did find examples elsewhere in our code so kept it exposed. But I've now fixed it.

Reply:
I mean this function should be under `#[cfg(test)]`, otherwise it can be misused in production and also require all these workarounds with dead and unused.

Reply:
hmm.. I was under the assumption code in `#[cfg(test)]` wasn't shared to lower modules. I'll give it a try and see if it works

Reply:
it might not, I'm not sure. But that's why I suggested moving everything to a separate module, something like `tests`